### PR TITLE
DC-31541 Don't overwrite nature_of_change in background jobs

### PR DESCRIPTION
**Pull request by ThrawnCA**
Background jobs have already been fixed so that they don't fail validation, but they also need to be adjusted so that they don't overwrite nature_of_change with a blank.

### CODE REVIEW

File: ckanext/data_qld/helpers.py [hunk 1]
```diff
@@ -21,6 +21,9 @@ def get_user():
 
 
 def user_has_admin_access(include_editor_access):
```
Comment (duttonw) [inline ckanext/data_qld/helpers.py:23]: time to rename this function. Its user_has_admin_access_or_background_task

**Review by duttonw — APPROVED** [review]

File: ckanext/data_qld/helpers.py (hunk 1 above)
Comment (ThrawnCA) [reply ckanext/data_qld/helpers.py:23]: I'm not so sure it needs a rename...background tasks are part of the system, therefore they do get admin access.

**Comment by ThrawnCA** [comment]
It turns out that ckanext-csrf-filter is a bit strict in detecting API requests (which are exempt from CSRF checks); the double leading slash `//api`, although it is accepted by CKAN and treated as a single slash, causes it to fail the regex and get treated as a normal POST.

**Comment by duttonw** [comment]
> It turns out that ckanext-csrf-filter is a bit strict in detecting API requests (which are exempt from CSRF checks); the double leading slash `//api`, although it is accepted by CKAN and treated as a single slash, causes it to fail the regex and get treated as a normal POST.

so are we going to 'improve' the regex to match ckan ?

**Comment by ThrawnCA** [comment]
> so are we going to 'improve' the regex to match ckan ?

Probably, at some point. That can happen after this pull request, though.

**Pull request merged**